From ecfe42aeee2551a4ba24929cd5d7c3e544b23b25 Mon Sep 17 00:00:00 2001
From: Timple Raj M <quic_timple@quicinc.com>
Date: Thu, 23 Nov 2023 23:10:49 +0530
Subject: [PATCH] xhci-trace: fix to resolve the warning in debug build

drivers/usb/host/./xhci-trace.h: In function
'trace_event_get_offsets_xhci_log_msg':
./include/trace/trace_events.h:256:27: error: function
'trace_event_get_offsets_xhci_log_msg' can never be inlined
because it uses variable argument lists
  256 | static inline notrace int trace_event_get_offsets_##call( \
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~
drivers/usb/host/./xhci-trace.h:28:1: note: in expansion of macro
'DECLARE_EVENT_CLASS'
   28 | DECLARE_EVENT_CLASS(xhci_log_msg,
      | ^~~~~~~~~~~~~~~~~~~

samples/trace_events/./trace-events-sample.h: In function
'trace_event_get_offsets_foo_bar':
./include/trace/trace_events.h:256:27: error: function
'trace_event_get_offsets_foo_bar' can never be inlined
because it uses variable argument lists
  256 | static inline notrace int trace_event_get_offsets_##call( \
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~
./include/trace/trace_events.h:40:9: note: in expansion of macro
'DECLARE_EVENT_CLASS'
   40 |         DECLARE_EVENT_CLASS(name, \
      |         ^~~~~~~~~~~~~~~~~~~
samples/trace_events/./trace-events-sample.h:277:1: note:
in expansion of macro 'TRACE_EVENT'
  277 | TRACE_EVENT(foo_bar,
      | ^~~~~~~~~~~

Change-Id: I11ec974323209d56cd17ebf5dd69b7c22c087dae
Signed-off-by: Timple Raj M <quic_timple@quicinc.com>
---
 drivers/usb/host/xhci-trace.h              | 2 +-
 samples/trace_events/trace-events-sample.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/xhci-trace.h b/drivers/usb/host/xhci-trace.h
index 61e93a3540a7..ec2437b2e1c6 100644
--- a/drivers/usb/host/xhci-trace.h
+++ b/drivers/usb/host/xhci-trace.h
@@ -28,7 +28,7 @@
 DECLARE_EVENT_CLASS(xhci_log_msg,
 	TP_PROTO(struct va_format *vaf),
 	TP_ARGS(vaf),
-	TP_STRUCT__entry(__vstring(msg, vaf->fmt, vaf->va)),
+	TP_STRUCT__entry(__dynamic_array(char, msg, XHCI_MSG_MAX)),
 	TP_fast_assign(
 		__assign_vstr(msg, vaf->fmt, vaf->va);
 	),
diff --git a/samples/trace_events/trace-events-sample.h b/samples/trace_events/trace-events-sample.h
index 1a92226202fc..64460d6de81b 100644
--- a/samples/trace_events/trace-events-sample.h
+++ b/samples/trace_events/trace-events-sample.h
@@ -288,7 +288,7 @@ TRACE_EVENT(foo_bar,
 		__dynamic_array(int,	list,   __length_of(lst))
 		__string(	str,	string			)
 		__bitmask(	cpus,	num_possible_cpus()	)
-		__vstring(	vstr,	fmt,	va		)
+		__dynamic_array(char,   vstr,	512		)
 	),
 
 	TP_fast_assign(
-- 
2.34.1

